src/symbol/projection.js

```diff
@@ -576,10 +560,7 @@ function placeGlyphAlongLine(
 
     if (dir < 0) angle += Math.PI;
 
-    let currentIndex = dir > 0 ?
-        lineStartIndex + anchorSegment :
-        lineStartIndex + anchorSegment + 1;
-
+    let currentIndex = lineStartIndex + anchorSegment + (dir > 0 ? 0 : 1) | 0;
```

Comment:
Is it a leftover to leave the `| 0`? Should this either be `(dir > 0) | 0` or just `(dir > 0 ? 0 : 1)`?

Reply:
It's a bit of an obscure optimization trick, but bitwise OR has a very low precedence so it acts on the whole value of `currentIndex`, and the purpose is to make sure it's treated as a "small integer" by the JS engine rather than a double — there are lots of object lookups with this as the key and it makes sure they're optimized, sometimes surprisingly making the code faster. This might not be necessary in this particular case but I wanted to be on the safe side because it's a hot path.

Reply:
It's a bit of an obscure optimization trick, but bitwise OR has a very low precedence so it acts on the whole value of `currentIndex`, and the purpose is to make sure it's treated as a "small integer" by the JS engine rather than a double — there are lots of object lookups with this as the key and it makes sure they're optimized, sometimes surprisingly making the code faster. This might not be necessary in this particular case but I wanted to be on the safe side because it's a hot path.